### PR TITLE
Fix delete/move watched folder

### PR DESCRIPTION
**Pull request by julianmesa-gitkraken**
To be consistent in the three platforms these commits will make to return the error:
`Service shutdown: root path changed (renamed or deleted)` 
when the watched folder is deleted or renamed.

### CODE REVIEW

**Review by ianhattendorf — CHANGES_REQUESTED** [review]
This looks good, could we add unit tests for this? We should at least test:
1. exclude a path, modify a file in that path -> ensure we don't get any events
2. rename + delete the root path -> ensure we get an error

File: src/linux/InotifyTree.cpp [hunk 1]
```diff
@@ -71,6 +78,9 @@ bool InotifyTree::getPath(std::string &out, int wd) {
 }
 
 bool InotifyTree::hasErrored() {
+  if (!existWatchedPath()) {
+    mError = "Service shutdown: root path changed (renamed or deleted)";
+  }
   return mError != "";
```
Comment (ianhattendorf) [inline src/linux/InotifyTree.cpp:84]: Same comment here as with `Watcher::getError()`

**Review by ianhattendorf — CHANGES_REQUESTED** [review]

**Comment by julianmesa-gitkraken** [comment]
> This looks good, could we add unit tests for this? We should at least test:
> 
> 1. exclude a path, modify a file in that path -> ensure we don't get any events

It will be done in a specific PR for excluded paths. This PR is to homogeinize the error when the watched folder is deleted or renamed.
> 2. rename + delete the root path -> ensure we get an error
>    a. what happens if we rename/delete the parent directory of the root? e.g. we're watching `/foo/bar` and we rename/delete `/foo` to `/baz`: do we also handle this correctly?

The code will catch it and throw the error.

Ok, I will do the next tests:
1. Rename watched path
2. Delete watched path
3. Rename parent of the watched path
4. Delete parent of the watched path

File: js/spec/index-spec.js [hunk 2]
```diff
@@ -587,6 +587,128 @@ describe('Node Sentinel File Watcher', function() {
...
+          assert.equal(errorMsg, 'Service shutdown: root path changed (renamed or deleted)');
+        } finally {
+          try {
+            await watch.stop();
+          } catch (err) { /* do nothing */ }
+          watch = null;
+        }
+      });
+
+      it.only('Report error on rename parent of the watched folder', async function() {
```
Comment (ianhattendorf) [inline js/spec/index-spec.js:648]: I'm assuming the `it.only` was for debugging and should be removed.

**Review by ianhattendorf — CHANGES_REQUESTED** [review]

**Review by ianhattendorf — APPROVED** [review]
Approving on the condition the above comment is addressed, go ahead and merge once that's taken care of.

File: js/spec/index-spec.js (hunk 2 above)
Comment (julianmesa-gitkraken) [reply js/spec/index-spec.js]: Yeah, I forgot to remove it. Fixed.

**Pull request merged**